include/flatbuffers/flatbuffers.h

```diff
@@ -515,6 +509,25 @@ class DetachedBuffer {
   size_t reserved_;
   uint8_t *cur_;
   size_t size_;
+
+  inline void destroy() {
+    if (buf_) {
+      assert(allocator_);
+      allocator_->deallocate(buf_, reserved_);
+    }
+    if (own_allocator_ && allocator_) {
+      delete allocator_;
+    }
+  }
+
+  inline void empty_assign(DetachedBuffer& other) {
```

Comment:
Suggestions for this method:
- Can we just make it operate on `this` and change the calls to `other.empty_assign()`?
- If we want to keep the naming similar to STL, the closest thing would probably be `std::unique_ptr<T>::release()`, so my vote would be to name it `release`.
- Can we make it private/protected, since it's rather unsafe for the outside world to use?

Reply:
It is protected.
Why call it release, it does not return any kind of detached buffer?
`reset` is much closer to it's functionality.

Reply:
Oops, sorry, misread the diff in the GitHub UI, thought it was immediately under `data()`.

The semantics of `reset` are different in my mind though, that destroys the held object vs `release` merely detaches it. At the end of the day I don't really care what it's called, I'll let @gwvo be the real reviewer.

I still think that we should be doing `other.foo()` rather than `this->food(other)`. Even if not, I think there's also something in the Google guidelines that says mutable references should be passed in as pointers rather than references.

Reply:
`reset` isn't ideal. It technically releases any existing buffer. However it does not return it.
So it's really just empty assigning with assumed destruction.